TSOClient/FSO.Server/DataService/Providers/ServerAvatarProvider.cs

```diff
@@ -333,6 +333,7 @@ private Avatar HydrateOne(DbAvatar dbAvatar, DbLot dbLot)
                 AvatarSkills_LockLv_Mechanical = dbAvatar.lock_mechanical
             };
             result.Avatar_PrivacyMode = dbAvatar.privacy_mode;
+            result.Avatar_IsMentor = DAFactory.Get().Users.GetMentorStatus(dbAvatar.user_id);
```

Comment:
HydrateOne shouldn't call out to the DAFactory, it should be passed all relevant database objects. This makes it easier to batch them if we need to (like we do with properties).